src/main/java/com/opencart/steps/SearchProductPageBL.java

```diff
@@ -30,17 +30,28 @@ public SearchProductPageBL makeSearch(String value) {
     }
 
     public void verifySearchResult(String searchCriteria) {
-        Assert.assertTrue(findAllProductBySearchCriteria(searchCriteria));
+        SoftAssert softAssert = new SoftAssert();
+        String expectedResult = "There is no product that matches the search criteria.";
+        String actualResult = searchProductResultPage.getNoProductsFoundTitle().getText();
+        softAssert.assertEquals(expectedResult, actualResult, "Search criteria is actually correct");
+        softAssert.assertTrue(findAllProductBySearchCriteria(searchCriteria),
+                "Check the list of products Done!");
+
+        /* Our test will fail with searchProductWithCorrectSearchCriteria()
+         * in SearchProductTest class when we use assertAll method
+         * softAssert.assertAll();
+         */
```

Comment:
It is not right when you use SoftAssert without .assertAll() that will produce false-positive tests (your assertion always will be passed). In this case, try to separate logic for successful and unsuccessful search results.